cmake/CCCLBuildCompilerTargets.cmake

```diff
@@ -65,6 +66,10 @@ function(cccl_build_compiler_targets)
     list(APPEND cxx_compile_definitions "CCCL_DISABLE_RTTI")
   endif()
 
+  if (CCCL_SUPPRESS_ICC_DEPRECATION_WARNING)
+    list(APPEND cxx_compile_definitions "CCCL_SUPPRESS_ICC_DEPRECATION_WARNING")
+  endif()
+
```

Comment:
I would rather not add another option to our ci machinery.

Below we have a specific if statement that adds warnings / flags for ICC.

I would rather add that there

Reply:
I am trying to follow @alliepiper's advice and suppress the warning from the CMake presets. AFAIK, I need a cmake option so the preset can set it.

Reply:
IIUC, you want me to unconditionally suppress the warning from our CMake, because our CMake compile definitions are only applied to tests targets? I have to check whether that is the case, but then we can also go that route.

Reply:
We discussed this internally and decided to go ahead with the cmake option to make it easier for users to avoid that warning